tests/components/mqtt/test_lock.py

```diff
@@ -131,19 +157,24 @@ async def test_controlling_non_default_state_via_topic(
     assert state.state is STATE_UNLOCKED
     assert not state.attributes.get(ATTR_ASSUMED_STATE)
 
-    async_fire_mqtt_message(hass, "state-topic", "closed")
+    async_fire_mqtt_message(hass, "state-topic", payload)
 
     state = hass.states.get("lock.test")
-    assert state.state is STATE_LOCKED
-
-    async_fire_mqtt_message(hass, "state-topic", "open")
-
-    state = hass.states.get("lock.test")
-    assert state.state is STATE_UNLOCKED
+    assert state.state is lock_state
 
 
+@pytest.mark.parametrize(
+    "payload,lock_state",
+    [
+        ('{"val":"JAMMED"}', STATE_JAMMED),
```

Comment:
Lets leave jammed out of the state here and add a new topic to receive a jammed condition. It will need a second payload/state NOT_JAMMED. You could also choose to add this with a separate PR, if that is more convenient.